base/src/main/java/com/blazebit/expression/base/LocalDateOperatorInterpreter.java

```diff
@@ -0,0 +1,116 @@
+    }
+
+    @Override
+    public Boolean interpret(ExpressionInterpreter.Context context, DomainType leftType, DomainType rightType,
+            Object leftValue, Object rightValue, ComparisonOperator operator) {
+        if (leftValue instanceof LocalDate && rightValue instanceof LocalDate) {
+            LocalDate l = (LocalDate) leftValue;
+            LocalDate r = (LocalDate) rightValue;
+            switch (operator) {
+            case EQUAL:
+                return l.compareTo(r) == 0;
+            case NOT_EQUAL:
+                return l.compareTo(r) != 0;
+            case GREATER_OR_EQUAL:
+                return l.compareTo(r) > -1;
+            case GREATER:
+                return l.compareTo(r) > 0;
+            case LOWER_OR_EQUAL:
+                return l.compareTo(r) < 1;
+            case LOWER:
+                return l.compareTo(r) < 0;
+            default:
+                break;
+            }
+        } else {
+            throw new DomainModelException("Illegal arguments [" + leftValue + ", " + rightValue + "]!");
+        }
+
+        throw new DomainModelException(
+                "Can't handle the operator " + operator + " for the arguments [" + leftValue + ", " + rightValue
+                        + "]!");
+    }
+
+    @Override
+    public Object interpret(ExpressionInterpreter.Context context, DomainType targetType, DomainType leftType,
+            DomainType rightType, Object leftValue, Object rightValue, DomainOperator operator) {
+        if (leftValue instanceof TemporalInterval && rightValue instanceof TemporalInterval) {
+            TemporalInterval interval1 = (TemporalInterval) leftValue;
+            TemporalInterval interval2 = (TemporalInterval) rightValue;
+
+            switch (operator) {
+            case PLUS:
+                return interval1.add(interval2);
+            case MINUS:
+                return interval1.subtract(interval2);
+            default:
+                break;
+            }
+        } else if (leftValue instanceof LocalDate && rightValue instanceof TemporalInterval) {
+            LocalDate localDate = (LocalDate) leftValue;
+            Instant instant = localDate.atStartOfDay(ZoneId.systemDefault()).toInstant();
+            TemporalInterval interval = (TemporalInterval) rightValue;
+            switch (operator) {
+            case PLUS:
+                return interval.add(instant);
+            case MINUS:
+                return interval.subtract(instant);
+            default:
+                break;
+            }
+        } else if (leftValue instanceof TemporalInterval && rightValue instanceof LocalDate) {
+            TemporalInterval interval = (TemporalInterval) leftValue;
+            LocalDate localDate = (LocalDate) rightValue;
+            Instant instant = localDate.atStartOfDay(ZoneId.systemDefault()).toInstant();
+
+            if (operator == DomainOperator.PLUS) {
+                return interval.add(instant);
+            }
```

Comment:
IMO we should add `LocalDate add(LocalDate)` method to `TemporalInterval` in the blaze-domain project: https://github.com/Blazebit/blaze-domain/blob/master/core/api/src/main/java/com/blazebit/domain/runtime/model/TemporalInterval.java